scipy/optimize/tests/test_least_squares.py

```diff
@@ -822,3 +822,32 @@ def chi2(x):
     # if we choose an initial condition that is close to the solution
     # we shouldn't return an answer that is further away from the solution
     assert_allclose(res.x, answer, atol=initial_guess-answer)
+
+
+def test_gh_19103():
+    # Checks that least_squares trf method selects a strictly feasible point,
+    # and thus succeeds instead of failing,
+    # when the initial guess is reported exactly at a boundary point.
+    # This is a reduced example from gh191303
+    np.random.seed(1)
```

Comment:
Maybe I'm missing something obvious, but can you remind me what the seed pinning is doing here?

If it does do something useful, is there any way to use the newer/preferred `default_rng` from https://numpy.org/doc/stable/reference/random/generator.html? 

If it isn't needed, we could just delete I think. Beyond that, I was able to confirm fail before/pass after with this test, but I'll try to wait longer before merging this time, since I merged the original apparent bug!

Reply:
Indeed not needed.